components/SplitPane/common/src/commonMain/kotlin/org/jetbrains/compose/animation/MovableInterface.kt

```diff
@@ -0,0 +1,11 @@
+package org.jetbrains.compose.animation
+
+interface SingleDirectionMoveScope {
+    fun moveBy(pixels: Float): Float
+}
+
+interface SingleDirectionMovable {

```

Comment:
I had use this interface in DefaultSingleDirectionMovableState implementation.
It is private class similar to DefaultScrollState.

There is only difference with scroll state and this interfaces that we didn't return unconsumed movement.
Cause at that moment I could not imagine any case with nested drag or nested movement.

Unfortunately this interfaces should be public or at least internal for components, cause everything that could move in single direction may want to implement them.

Like, if somebody (khm me khm) want to implement brush with handles [example](https://observablehq.com/@d3/brush-handles), he possibly will implement them

Reply:
> I had use this interface in DefaultSingleDirectionMovableState implementation.

But if we don't use methods of this interface, and only implement it, there is no point in interface:
https://github.com/JetBrains/compose-jb/commit/7c20486df7bbffd3624098325ab18c354afcd9d0

Can you show use case of this interface for external users?

> at least internal for components

I think, if we make it internal, and there is only one implementation, it is better to get rid of interface too, it will be easier to read.

We can add interface at any moment, if we will need it in the future (for external users, or for tests).

Reply:
Well the reason of it existence same as for scrolling, somebody could implement `SingleDirectionMovable.smoothMoveBy()` animation.

But you are right at that moment there is only one implementation of this interface so possibly we could get rid of it.
At least until there will be at least few of implementations.